ibm/resource_ibm_network_gateway.go

```diff
@@ -0,0 +1,816 @@
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"ipv6_address_id": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+
+			// SoftLayer does not support public_ipv6_subnet configuration in vm creation. So, public_ipv6_subnet
+			// is defined as a computed parameter.
+			"public_ipv6_subnet": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"vlan_number": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+			"private_network_only": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+				ForceNew: true,
+			},
+		},
+	}
+}
+
+func resourceIBMNetworkGatewayCreate(d *schema.ResourceData, meta interface{}) error {
+	sess := meta.(ClientSession).SoftLayerSession()
+	hwService := services.GetHardwareService(sess)
+	var order datatypes.Container_Product_Order
+	var err error
+	quote_id := d.Get("quote_id").(int)
+	hardware := datatypes.Hardware{
+		Hostname: sl.String(d.Get("hostname").(string)),
+		Domain:   sl.String(d.Get("domain").(string)),
+	}
+
+	if quote_id > 0 {
+		// Build a Network Gateway from a quote.
+		order, err = services.GetBillingOrderQuoteService(sess).
+			Id(quote_id).GetRecalculatedOrderContainer(nil, sl.Bool(false))
+		if err != nil {
+			return fmt.Errorf(
+				"Encountered problem trying to get the Network Gateway order template from quote: %s", err)
+		}
+		order.Quantity = sl.Int(1)
+		order.Hardware = make([]datatypes.Hardware, 0, 1)
+		order.Hardware = append(
+			order.Hardware,
+			hardware,
+		)
+	} else {
+		// Build a montly Network gateway
+		order, err = getMonthlyGatewayOrder(d, meta)
+		if err != nil {
+			return fmt.Errorf(
+				"Encountered problem trying to get the Gateway order template: %s", err)
+		}
+	}
+
+	order, err = setCommonGatewayOrderOptions(d, meta, order)
+	if err != nil {
+		return fmt.Errorf(
+			"Encountered problem trying to configure Gateway options: %s", err)
+	}
+
+	var ProductOrder datatypes.Container_Product_Order
+	ProductOrder.OrderContainers = make([]datatypes.Container_Product_Order, 1)
+	ProductOrder.OrderContainers[0] = order
+
+	_, err = services.GetProductOrderService(sess).VerifyOrder(&ProductOrder)
+	if err != nil {
+		return fmt.Errorf(
+			"Encountered problem trying to verify the order: %s", err)
+	}
+	_, err = services.GetProductOrderService(sess).PlaceOrder(&ProductOrder, sl.Bool(false))
+	if err != nil {
+		return fmt.Errorf(
+			"Encountered problem trying to place the order: %s", err)
+	}
+
+	log.Printf("[INFO] Gateway ID: %s", d.Id())
+
+	bm, err := waitForNetworkGatewayProvision(&hardware, meta)
+	if err != nil {
+		return fmt.Errorf(
+			"Error waiting for Gateway (%s) to become ready: %s", d.Id(), err)
+	}
+
+	id := *bm.(datatypes.Hardware).Id
+	d.SetId(fmt.Sprintf("%d", id))
+
+	// Set tags
+	err = setHardwareTags(id, d, meta)
+	if err != nil {
+		return err
+	}
+
+	var storageIds []int
+	if storageIdsSet := d.Get("file_storage_ids").(*schema.Set); len(storageIdsSet.List()) > 0 {
```

Comment:
`file_storage_ids` and `block_storage_ids` are not defined on the schema.

Reply:
Removed and added the following reference to the documentation to avoid confusion:
* `RAID_CONTROLLER` - (reference) this version support a single RAID 1 controller pre configured with two hard drives 1 TB SATA drives